pkg/core/resources/apis/meshmultizoneservice/api/v1alpha1/meshmultizoneservice.go

```diff
@@ -21,14 +21,15 @@ type MeshMultiZoneService struct {
 	Selector Selector `json:"selector"`
 	// Ports is a list of ports from selected MeshServices
 	// +kubebuilder:validation:MinItems=1
-	Ports []Port `json:"ports,omitempty"`
+	Ports []Port `json:"ports"`
 }
 
 type Port struct {
-	Name string `json:"name,omitempty"`
-	Port uint32 `json:"port"`
+	Name *string `json:"name,omitempty"`
+	Port uint32  `json:"port"`
+	// +kubebuilder:validation:Optional
 	// +kubebuilder:default=tcp
-	AppProtocol core_mesh.Protocol `json:"appProtocol,omitempty"`
+	AppProtocol core_mesh.Protocol `json:"appProtocol"`
```

Comment:
Looks like we're missing proper validation on MMZS from the beginning, potentially `appProtocol: ""` can be introduced in the code here https://github.com/kumahq/kuma/blob/74af069803862ec569ed5ffe43974ee6b092b274/pkg/plugins/policies/core/xds/meshroute/listeners.go#L213

Technically it was like that before your change, so I don't know maybe it should be a separate issue.